### PR TITLE
SW-2624 Use New Filters in Species Table

### PR DESCRIPTION
**Pull request by charlie-tf**
image
image
image

### CODE REVIEW

File: src/components/Species/index.tsx [hunk 1]
```diff
@@ -276,6 +292,62 @@ export default function SpeciesList({ reloadData, species }: SpeciesListProps):
...
+              type: 'Exact',
+              values: [selectedOrganization.id],
+            },
+          ],
+        },
+        sortOrder: [{ field: 'id', direction: 'Ascending' }],
+        count: 1000,
+      };
+
+      const data = await SearchService.search(searchParams);
```
Comment (karthikbtf) [inline src/components/Species/index.tsx:328]: It's likely I missed this before.. there's an endpoint that retrieves values for dropdowns. See how we do this for accessions table https://github.com/terraware/terraware-web/blob/main/src/services/SeedBankService.ts#L142 may need some refactor to move it out into search.

File: src/components/Species/index.tsx [hunk 2]
```diff
@@ -750,12 +805,40 @@ export default function SpeciesList({ reloadData, species }: SpeciesListProps):
...
+              }}
+              transformOrigin={{
+                vertical: 'top',
+                horizontal: 'center',
+              }}
+              className={classes.popoverContainer}
+            >
+              <FilterGroup
+                initialFilters={filters}
+                fields={filterColumns}
```
Comment (karthikbtf) [inline src/components/Species/index.tsx:828]: thoughts on hiding (or marking the dropdowns as disabled) if there are no values to select from? maybe not an issue in production.. in staging we have orgs with dummy species.. a lot of the filter dropdowns have no values to select from... ofc the data is mostly test data.

**Review by karthikbtf — APPROVED** [review]
looks good, some comments but totally fine if you want to address separately (if they need addressing)

File: src/components/Species/index.tsx (hunk 1 above)
Comment (charlie-tf) [reply src/components/Species/index.tsx:328]: I'll put that as a followup! Thanks

File: src/components/Species/index.tsx [hunk 3]
```diff
@@ -47,6 +53,9 @@ import { DropdownItem, SortOrder } from '@terraware/web-components';
 import { useLocalization, useOrganization } from 'src/providers/hooks';
 import { PillList, PillListItem } from '@terraware/web-components';
 import { isTrue } from 'src/utils/boolean';
+import FilterGroup, { FilterField } from '../common/FilterGroup';
+import { FieldOptionsMap } from '../../services/NurseryWithdrawalService';
+import SearchService from '../../services/SearchService';
```
[the remark this answers is not in the document]
Comment (charlie-tf) [reply src/components/Species/index.tsx:58]: oops yes

File: src/components/Species/index.tsx (hunk 2 above)
Comment (charlie-tf) [reply src/components/Species/index.tsx:828]: That would make sense. I'll look at doing that

**Review by karthikbtf — APPROVED** [review]

**Comment by karthikbtf** [comment]
Looks good in vercel, thanks so much 🙏

**Pull request merged**